log/logger_test.go

```diff
@@ -273,7 +270,8 @@ func TestCloseLogger(t *testing.T) {
 
```

Comment:
I think the new changes make `TestCloseLogger` take us off to the races
```
panic: send on closed channel [recovered]
	panic: send on closed channel

goroutine 46 [running]:
testing.tRunner.func1.2({0x2592c0, 0x2ac920})
	C:/Program Files/Go/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
	C:/Program Files/Go/src/testing/testing.go:1392 +0x39f
panic({0x2592c0, 0x2ac920})
	C:/Program Files/Go/src/runtime/panic.go:838 +0x207
github.com/thrasher-corp/gocryptotrader/log.(*multiWriterHolder).StageLogEvent(0xc000246300, 0xc0002500a8, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...)
	C:/Users/gloriousCode/go/src/github.com/thrasher-corp/gocryptotrader/log/logger_multiwriter.go:112 +0x1b9
github.com/thrasher-corp/gocryptotrader/log.TestMultiWriterWrite(0xc0002fa820)
	C:/Users/gloriousCode/go/src/github.com/thrasher-corp/gocryptotrader/log/logger_test.go:179 +0x1c8
testing.tRunner(0xc0002fa820, 0x284db0)
	C:/Program Files/Go/src/testing/testing.go:1439 +0x102
created by testing.(*T).Run
	C:/Program Files/Go/src/testing/testing.go:1486 +0x35f
```
While I think this is a result of a test setup issue, just be careful its not a logger race issue

Reply:
Stage log event is usually called by getting log fields first which are protected. In that specific function, we are bypassing that protection. So outside of this test function, it shouldn't be called unless we export it later. Have moved close logger after the test functions have finished.